scipy/stats/_stats_py.py

```diff
@@ -710,13 +710,22 @@ def tmin(a, lowerlimit=None, axis=0, inclusive=True, nan_policy='propagate'):
     14
 
     """
+    xp = array_namespace(a)
+
+    # remember original dtype; _put_val_to_limits might need to change it
     dtype = a.dtype
-    a, _ = _put_val_to_limits(a, (lowerlimit, None), (inclusive, None))
-    res = np.nanmin(a, axis=axis)
-    if not np.any(np.isnan(res)):
+    a, mask = _put_val_to_limits(a, (lowerlimit, None), (inclusive, None),
+                                 val=xp.inf, xp=xp)
+
+    min = xp.min(a, axis=axis)
+    n = xp.sum(xp.asarray(~mask, dtype=a.dtype), axis=axis)
+    res = xp.where(n != 0, min, xp.nan)
```

Comment:
There is no `nanmin` or similar in the standard, so instead of replacing trimmed values with NaN, we replace with `inf`, which won't change the minimum. If *all* values were trimmed, the result is NaN (like before).